### ISSUE TITLE
JTReg Test Failure: jdk/internal/reflect/constantPool/ConstantPoolTest.java

### ISSUE THREAD

**Issue by adam-thorpe** [eclipse/openj9#7249]
Failure link
------------
rebuild: https://ci.adoptopenjdk.net/job/Grinder/parambuild/?JDK_VERSION=11&JDK_IMPL=openj9&BUILD_LIST=openjdk&JenkinsFile=openjdk_x86-64_linux&CUSTOM_TARGET=test/jdk/jdk/internal/reflect/constantPool/ConstantPoolTest.java&TARGET=jdk_custom_0
- test category: openjdk
- OS/architecture: all
- version: 11, 13
- JVM: openj9 only

Optional info
-------------
- intermittent failure (yes|no): no
- regression or new test: regression

Failure output (captured from console output)
---------------------------------------------
```
TEST RESULT: Error. Agent communication error: java.io.EOFException; check console log for any additional details
```
Test doesn't throw any sort of error other than this which isn't exactly helpful.

**Comment by smlambert** [comment]
https://github.com/AdoptOpenJDK/openjdk-tests/pull/1369

**Comment by DanHeidinga** [comment]
@adam-thorpe Do you have the console log?  Is there anything relevant in it?

Also, can you link to the source for the test?  It will make it easier for whoever is looking at the issue to get started.

**Comment by adam-thorpe** [comment]
No, but should be easy to recreate with the jenkins link above. Test here: https://github.com/AdoptOpenJDK/openjdk-jdk11u/blob/master/test/jdk/jdk/internal/reflect/constantPool/ConstantPoolTest.java

**Comment by DanHeidinga** [comment]
https://github.com/AdoptOpenJDK/openjdk-jdk11u/blob/9456afc1a52221bb397bf067ca39e90dee7411ca/test/jdk/jdk/internal/reflect/constantPool/ConstantPoolTest.java#L83-L89

This looks like the test is expecting particular CP entries at statically known indexes.  OpenJ9 reorders the constantpool as it converts from classfiles to J9ROMClasses.  Assuming my reading of the test is correct, this won't apply to OpenJ9.

**Comment by JasonFengJ9** [comment]
From an internal JDK18 build `job/Test_openjdknext_j9_sanity.openjdk_x86-64_linux_Personal/128/consoleFull`

```
19:37:11  openjdk version "18-internal" 2022-03-15
19:37:11  OpenJDK Runtime Environment (build 18-internal+0-adhoc.jenkins.BuildJDKnextx86-64linuxPersonal)
19:37:11  Eclipse OpenJ9 VM (build master-5afc033b29e, JRE 18 Linux amd64-64-Bit Compressed References 20211206_914 (JIT enabled, AOT enabled)
19:37:11  OpenJ9   - 5afc033b29e
19:37:11  OMR      - 5e9418dcfde
19:37:11  JCL      - 06ac806d8f8 based on jdk-18+24)

20:45:57  --------------------------------------------------
20:46:12  TEST: jdk/internal/reflect/constantPool/ConstantPoolTest.java

20:46:12  stderr:
20:46:12  01:45:56.497 0x7f8838752300   j9jcl.347    *   ** ASSERTION FAILED ** at /home/jenkins/workspace/Build_JDKnext_x86-64_linux_Personal/openj9/runtime/jcl/common/sun_reflect_ConstantPool.c:833: ((0 ))
```

This test has been excluded for early levels, need to add it to JDK18 as well.
